### ISSUE TITLE
C# Debug not reading correct lines of code

### ISSUE THREAD

**Issue by ondesic** [godotengine/godot#42258]
I am using 3.2.3 stable on Windows 10 with C#

After moving from 3.2.2 stable to 3.2.3 stable, I added some new code to a .cs file.  However, Debug is not recognizing new code added to a .cs file.  Somehow it is going off of a memory of the old code.  
In the example below the breakpoint is set to line 234.  However, the code is reading as if it is on line 252:
![first](https://user-images.githubusercontent.com/26915986/93964450-d4d71b00-fd1c-11ea-9bc9-531025b3a058.gif)
The local Tabs variable "m" has already been assigned even though I have not made it  to this line yet.
In the next picture, after stepping down 2 lines to 236, the Tabs variable "it" is now assigned.
![second](https://user-images.githubusercontent.com/26915986/93964608-48792800-fd1d-11ea-81ac-d9f290593548.gif)

I have cleaned and rebuilt the project with no luck.  I haven't encountered this before.

**Comment by ondesic** [comment]
Doing some reading it could be because the Optimizer setting is not set to false.
in the old .csproj file, under ` 'ExportDebug|AnyCPU'` the following line is included: `"Optimize>false</Optimize>"`
Under 'ExportRelease|AnyCPU' Optimizer is set to TRUE.

The new .csproj does not have this setting and so it keeps deleting the changes I make.  How do I add the Optimize bool back in?

**Comment by Calinou** [comment]
cc @neikeq

**Comment by ondesic** [comment]
Any thoughts on this?  I can't add any code in the current system and so I'm stuck.

**Comment by ondesic** [comment]
Two things.  
1) I am trying to figure this out.  I have searched through the project files and setting and cannot find the "Optimize" setting in any of the project files. 
 This setting used to be in the .csproj file in 3.2.2's and earlier.
2) I am changing the name of this post to "New code not being recognized on run".  I don't believe this is a debug problem.  It seems that Godot is using a copy of my code from earlier.  This happens when Optimize is set to TRUE.

**Comment by akien-mga** [comment]
Does it work if you delete/rename the `res://.mono` folder to force it to be recreated? (Maybe make a copy before deleting it, in case you need to restore it or share it for debugging purposes.)

**Comment by ondesic** [comment]
Just did that and then Cleaned and restored.  None of the signals now connect.

**Comment by ondesic** [comment]
I just can't figure this out.  I have tried so many different things with no luck.  I wish there was someone I could pay to spend an hour with me to figure this out, but it is what it is.

**Comment by neikeq** [comment]
I will be doing some work on Windows in a few days and I will check this.

**Comment by ondesic** [comment]
@akien-mga @neikeq, For almost two months I have gone back to 3.2.2 because none of the future releases/betas have allowed me to make changes to my code.
I downloaded 3.2.4 beta 2 and was sad to find the same problem. My 3.2.2 project runs in 3.2.4 beta 2, but cannot be altered.  It is stuck with using the C# code that was present when I opened the project for the first time in 3.2.4 beta 2.
I have got to figure this out or I won't be able to use anything after 3.2.2
Any more ideas?

**Comment by ondesic** [comment]
Today I created a project in 3.2.2.  I then opened it in 3.2.4 beta 2 to let Godot convert it.
I went file by file and compared my project to the new project.
I found that in the .csproj file there were a few extra lines in my big project that are not in the new project.
the lines were:
`<RuntimeIdentifier>win</RuntimeIdentifier>`
`<LangVersion>7.1</LangVersion>`

I erased these lines and the project is now updating correctly.  I have no idea why, but perhaps this will help someone down the road.
